Include/monitoring.h

```diff
@@ -0,0 +1,276 @@
+#ifndef Py_MONITORING_H
+#define Py_MONITORING_H
+#ifdef __cplusplus
+extern "C" {
+
+#endif
+
+typedef struct _PyMonitoringState {
+    uint8_t active;
+    uint8_t opaque;
+} PyMonitoringState;
+
+
+#ifndef Py_LIMITED_API
+#  define Py_CPYTHON_MONITORING_H
+#  include "cpython/monitoring.h"
+#  undef Py_CPYTHON_MONITORING_H
+
+#define IF_ACTIVE(X)  \
+    if (state->active) { \
+        return (X); \
+    } \
+    else { \
+        return 0; \
+    }
+
+#endif
```

Comment:
I'm not entirely happy with this macro. It leaks into user code as an unescaped global C name, and it makes it rather opaque what the decision depends on (`state` is not passed in). I understand that you want to extract the boring decision taking, but then I'd rather keep it local (and duplicated) to the internal usages, define it as `_PyMonitoring_IF_ACTIVE(state, event_call)`, and undef it after its use.